### PR TITLE
[NFC] Retire ROCm < 3.7

### PR DESCRIPTION
**Pull request by atamazov**
- Removed HCC support from CMake files (except fin and composable_kernels)
- Removed HCC support from source code
- Removed HipCompilerVersion() - it was necessary for HCC only.
- Removed external_tool_version_t - it became useless
- WORKAROUND_SWDEV_255735 reworked to a fix
- Removed WORKAROUND_MLOPEN_ISSUE_2335 - not relevant since 3.7
  - See https://github.com/AMDComputeLibraries/MLOpen/issues/2335
- Removed WORKAROUND_SWDEV_200782 - not relevant since 3.7
- Removed WORKAROUND_SWDEV_220166 - not relevant since 3.7
- Removed some checks -- always true since 3.7
- Some formatting

### CODE REVIEW

**Comment by atamazov** [comment]
CI passed, please review

**Review by junliume — DISMISSED** [review]
Current, public support matrix does not extend to legacy ROCm before 3.7. I think it would be a good maintenance move and I will check on other alignments simultaneously.

File: src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops_padded_gemm.cpp [hunk 1]
```diff
@@ -36,11 +36,10 @@
 /// issues are not reproducible with ROCm no-npi build 6738-STG2 (proposed 4.2 release candidate).
 /// Let's disable these configs for 3.7...4.0.
 #define WORKAROUND_MI100_CONV_IMPLICIT_GEMM_HIP_FWD_V4R4_PADDED_GEMM_XDLOPS \
-    (HIP_PACKAGE_VERSION_FLAT >= 3007000000 && HIP_PACKAGE_VERSION_FLAT <= 4000999999)
+    (HIP_PACKAGE_VERSION_FLAT <= 4000999999)
```
Comment (averinevg) [inline src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops_padded_gemm.cpp:39]: To be consistent with other code
```suggestion
    (HIP_PACKAGE_VERSION_FLAT <= 4000999999ULL)
```

Comment (atamazov) [reply src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops_padded_gemm.cpp:39]: Right, thanks!

Comment (atamazov) [reply src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops_padded_gemm.cpp]: Resolved

**Comment by atamazov** [comment]
CI passed, please review

**Review by averinevg — APPROVED** [review]
LGTM

**Pull request merged**